src/common/com/intellij/plugins/haxe/ide/HaxeFileTemplateUtil.java

```diff
@@ -39,28 +40,28 @@
 public class HaxeFileTemplateUtil {
   private final static String HAXE_TEMPLATE_PREFIX = "Haxe";
 
-  public static List<FileTemplate> getApplicableTemplates() {
+  public static List<FileTemplate> getApplicableTemplates(Project project) {
     return getApplicableTemplates(new Condition<FileTemplate>() {
       @Override
       public boolean value(FileTemplate fileTemplate) {
         return HaxeFileType.DEFAULT_EXTENSION.equals(fileTemplate.getExtension());
       }
-    });
+    }, project);
   }
 
-  public static List<FileTemplate> getNMMLTemplates() {
+  public static List<FileTemplate> getNMMLTemplates(Project project) {
     return getApplicableTemplates(new Condition<FileTemplate>() {
       @Override
       public boolean value(FileTemplate fileTemplate) {
         return NMMLFileType.DEFAULT_EXTENSION.equals(fileTemplate.getExtension());
       }
-    });
+    }, project);
   }
 
-  public static List<FileTemplate> getApplicableTemplates(Condition<FileTemplate> filter) {
+  public static List<FileTemplate> getApplicableTemplates(Condition<FileTemplate> filter, Project project) {
     List<FileTemplate> applicableTemplates = new SmartList<FileTemplate>();
-    applicableTemplates.addAll(ContainerUtil.findAll(FileTemplateManager.getInstance().getInternalTemplates(), filter));
-    applicableTemplates.addAll(ContainerUtil.findAll(FileTemplateManager.getInstance().getAllTemplates(), filter));
+    applicableTemplates.addAll(ContainerUtil.findAll(FileTemplateManager.getInstance(project).getInternalTemplates(), filter));
```

Comment:
Yes, Srikanth is reviewing it right now, and I'm going to pull it later.  Then I'll kick off new builds for your and Elias' (Ilya's) changes and make sure they all build correctly.  (The unit tests for v15 do not run correctly, but that's a JetBrains bug that they are working on.  We'll re-enable running them after their fix.)